src/main/kotlin/com/terraformation/backend/funder/db/PublishedProjectDetailsStore.kt

```diff
@@ -0,0 +1,177 @@
+package com.terraformation.backend.funder.db
+
+import com.terraformation.backend.accelerator.model.CarbonCertification
+import com.terraformation.backend.accelerator.model.SustainableDevelopmentGoal
+import com.terraformation.backend.auth.currentUser
+import com.terraformation.backend.db.asNonNullable
+import com.terraformation.backend.db.default_schema.LandUseModelType
+import com.terraformation.backend.db.default_schema.ProjectId
+import com.terraformation.backend.db.funder.tables.records.PublishedProjectDetailsRecord
+import com.terraformation.backend.db.funder.tables.references.PUBLISHED_PROJECT_CARBON_CERTS
+import com.terraformation.backend.db.funder.tables.references.PUBLISHED_PROJECT_DETAILS
+import com.terraformation.backend.db.funder.tables.references.PUBLISHED_PROJECT_LAND_USE
+import com.terraformation.backend.db.funder.tables.references.PUBLISHED_PROJECT_SDG
+import com.terraformation.backend.funder.event.FunderProjectProfilePublishedEvent
+import com.terraformation.backend.funder.model.FunderProjectDetailsModel
+import jakarta.inject.Named
+import java.math.BigDecimal
+import java.time.InstantSource
+import org.jooq.DSLContext
+import org.springframework.context.ApplicationEventPublisher
+
+@Named
+class PublishedProjectDetailsStore(
+    private val clock: InstantSource,
+    private val dslContext: DSLContext,
+    private val eventPublisher: ApplicationEventPublisher,
+) {
+  fun fetchOneById(projectId: ProjectId): FunderProjectDetailsModel? {
+    // convert the list to a set
+    val sdgList =
+        with(PUBLISHED_PROJECT_SDG) {
+          dslContext
+              .select(SDG_NUMBER.asNonNullable())
+              .from(this)
+              .where(PROJECT_ID.eq(projectId))
+              .fetchSet(SDG_NUMBER.asNonNullable())
+              .mapNotNull { SustainableDevelopmentGoal.bySdgNumber[it] }
+              .toSet()
+        }
+    val carbonCerts =
+        with(PUBLISHED_PROJECT_CARBON_CERTS) {
+          dslContext
+              .select(CARBON_CERTIFICATION.asNonNullable())
+              .from(this)
+              .where(PROJECT_ID.eq(projectId))
+              .fetchSet(CARBON_CERTIFICATION.asNonNullable())
+              .mapNotNull { CarbonCertification.forDisplayName(it) }
+              .toSet()
+        }
+    val landUseModelMap =
+        with(PUBLISHED_PROJECT_LAND_USE) {
+          dslContext
+              .select(LAND_USE_MODEL_TYPE_ID, LAND_USE_MODEL_HECTARES)
+              .from(this)
+              .where(PROJECT_ID.eq(projectId))
+              .fetchMap(
+                  LAND_USE_MODEL_TYPE_ID.asNonNullable(), LAND_USE_MODEL_HECTARES.asNonNullable())
+        }
+    return with(PUBLISHED_PROJECT_DETAILS) {
+      dslContext
+          .selectFrom(this)
+          .where(PROJECT_ID.eq(projectId))
+          .fetchOneInto(PublishedProjectDetailsRecord::class.java)
```

Comment:
Is the class name needed here? I'd expect `.fetchOne()` to return a `PublishedProjectDetailsRecord`.

Reply:
Ah yes, I'll fix that. I originally had a different class in there but decided to move to the record.